mainhandler/vulnscan.go

```diff
@@ -94,143 +96,124 @@ func convertImagesToWebsocketScanCommand(registry *registryScan, sessionObj *uti
+		return fmt.Errorf("parseRegistryCommand failed with err %v", err)
 	}
 
+	err = actionHandler.testRegistryConnect(&registryScan, sessionObj)
 	if err != nil {
-		// if configmap not found, it means we will use all images and default depth
-		if strings.Contains(err.Error(), fmt.Sprintf("reason: configmaps \"%v\" not found", registryScanConfigmap)) {
-			glog.Infof("configmap: %s does not exists, using default values", registryScanConfigmap)
-			return NewRegistryScanConfig(registryName), string(cmDefaultMode), nil
-		} else {
-			return nil, string(cmDefaultMode), err
-		}
+		glog.Errorf("in testRegistryConnectivity: testRegistryConnect failed with error: %v", err.Error())
+		return err
 	}
-	configData := configMap.GetData()
-	var registriesConfigs []registryScanConfig
-	registriesConfigStr, ok := configData["registries"].(string)
+	return nil
+}
+
+func (actionHandler *ActionHandler) parseRegistryNameArg(sessionObj *utils.SessionObj) (string, error) {
+	registryInfo, ok := sessionObj.Command.Args[armotypes.RegistryInfoArgKey].(map[string]interface{})
 	if !ok {
-		return nil, string(cmDefaultMode), fmt.Errorf("error parsing %v confgimap: registries field not found", registryScanConfigmap)
-	}
-	registriesConfigStr = strings.Replace(registriesConfigStr, "\n", "", -1)
-	err = json.Unmarshal([]byte(registriesConfigStr), &registriesConfigs)
-	if err != nil {
-		return nil, string(cmDefaultMode), fmt.Errorf("error parsing ConfigMap: %s", err.Error())
+		return "", fmt.Errorf("could not parse registry info")
 	}
-	for _, config := range registriesConfigs {
-		if config.Registry == registryName {
-			return &config, string(cmLoadedMode), nil
-		}
+	registryName, ok := registryInfo[registryNameField].(string)
+	if !ok {
+		return "", fmt.Errorf("could not parse registry name")
 	}
-	return NewRegistryScanConfig(registryName), string(cmDefaultMode), nil
 
+	sessionObj.Reporter.SetTarget(fmt.Sprintf("%s: %s", apitypes.AttributeRegistryName,
+		registryName))
+	sessionObj.Reporter.SendDetails(fmt.Sprintf("registryInfo parsed: %v", registryInfo), true, sessionObj.ErrChan)
+	return registryName, nil
 }
 
-func (actionHandler *ActionHandler) getRegistryScanSecret() (k8sinterface.IWorkload, error) {
-	secret, err := actionHandler.k8sAPI.GetWorkload(armotypes.KubescapeNamespace, "Secret", armotypes.RegistryScanSecretName)
-	if err == nil && secret != nil {
-		return secret, err
+func (actionHandler *ActionHandler) testRegistryConnect(registry *registryScan, sessionObj *utils.SessionObj) error {
+	repos, err := registry.enumerateRepoes()
+	if err != nil {
+		if strings.Contains(strings.ToLower(err.Error()), "unauthorized") || strings.Contains(err.Error(), "DENIED") || strings.Contains(err.Error(), "authentication") {
+			// registry info is good, but authentication failed
+			sessionObj.Reporter.SetDetails(string(testRegistryInformationStatus))
+			sessionObj.Reporter.SendStatus(reporterlib.JobSuccess, true, sessionObj.ErrChan)
```

Comment:
how come error and job success together?

Reply:
information - success
authentication - failed